src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -6191,9 +6195,13 @@ void Assembler::push(int32_t imm32) {
   emit_int32(imm32);
 }
 
-void Assembler::push(Register src) {
-  int encode = prefix_and_encode(src->encoding());
-  emit_int8(0x50 | encode);
+void Assembler::push(Register src, bool is_pair) {
+  if (is_pair && VM_Version::supports_apx_f()) {
```

Comment:
Please move it to macroassembler.

Reply:
Please see the push and pop functions moved to macroassembler in the updated code.